40483 heading selectors (#40649)
mdn/content: files/en-us/web/css/_colon_heading_function/index.md

## Syntax

```css-nolint
:heading([ <An+B> [, <An+B>]* | even | odd ]) {
  /* ... */
}
```

### Parameters

The `:heading()` pseudo-class function takes a comma-separated list of `An+B` or the keyword values that describes a pattern for matching heading element.

#### Keyword values

- `odd`
  - : Represents the heading elements whose numeric position is odd: `<h1>`, `<h3>`, and `<h5>`.
- `even`
  - : Represents the heading elements whose numeric position is even: `<h2>`, `<h4>`, and `<h6>`.

#### Functional notation

- `<An+B>`
  - : Represents the heading elements whose numeric position matches the pattern `An+B`, for every positive integer or zero value of `n`, where:
    - `A` is an integer step size,
    - `B` is an integer offset,
    - `n` is all nonnegative integers, starting from 0.

    It can be read as the `An+B`-th element of a list. The `A` and `B` must both have {{cssxref("&lt;integer&gt;")}} values.

## Usage notes

> The `:heading()` functional pseudo-class only matches heading-level elements and not elements with a [`role="heading"`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/heading_role) attribute nor will it respect the ['aria-level'](/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-level) ARIA attribute.

Reviewer: ```suggestion
The `:heading()` functional pseudo-class matches only elements that are semantically recognized as headings. It does not match elements with a [`role="heading"`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/heading_role) attribute, nor does it respect the ['aria-level'](/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-level) ARIA attribute.
```

## Examples

### Using keyword parameters

In this example, the `odd` keyword matches headings with odd-numbered levels, so `<h1>` and `<h3>`. The `even` keyword is used to target even-numbered heading levels, `<h2>` and `<h4>`.

```html
<h1>Heading 1</h1>
<h2>Heading 2</h2>
<h3>Heading 3</h3>
<h4>Heading 4</h4>
```

```css
:heading(odd) {
  color: tomato;
}
:heading(even) {
  color: slateblue;
}
```

{{EmbedLiveSample("Keyword_example", "", "215")}}

### Using the `An+B` notation

```html
<h1>Science</h1>
<h2>Physics</h2>
<h3>Atomic, molecular, and optical physics</h3>
<h4>Optical physics</h4>
<h5>X-Rays</h5>
<h6>Discovery</h6>
```

```css hidden
main {
  display: flex;
  justify-content: space-around;
}
```

> ```css
> /* targeting headings <h3> & <h4> */
> :heading(3, 4) {
>   font-weight: 100;
> }
> /* targeting heading 3 and the preceding headings <h3>, <h2>, & <h1> */
> :heading(-n + 3) {
>   color: tomato;
> }
> /* targeting every 3rd heading-level form the first <h1>, & <h4> */
> :heading(3n + 1) {
>   font-style: italic;
> }
> /* targeting all heading-levels past 5 <h5>, & <h6> */
> :heading(n + 5) {
>   color: slateblue;
> }
> ```

Reviewer: - just a slight tweak in the tense
- also, since we have a more detailed explanation in the prose, i went with just including the logic in the comments in the code. The prose explains the exact heading levels that'll be selected. WDYT?
```suggestion
```css
/* Targets headings <h3> and <h4> */
:heading(3, 4) {
  font-weight: 100;
}
/* Targets headings in reverse starting from <h3> */
:heading(-n + 3) {
  color: tomato;
}
/* Targets every third heading starting from <h1> */
:heading(3n + 1) {
  font-style: italic;
}
/* Targets headings after level 5 */
:heading(n + 5) {
  color: slateblue;
}
```
```

---
title: :heading()
slug: Web/CSS/:heading_function
page-type: css-pseudo-class
browser-compat: css.selectors.headingfunction
sidebar: cssref
---

The **`:heading()`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) function matches [heading elements](/en-US/docs/Web/HTML/Reference/Elements/Heading_Elements) that match the functional notation. In other words, the `:heading()` functional pseudo-class allows you to style all headings that match the functional notation at once, rather than individually.

> [!NOTE]

> > The `:heading()` pseudo-class has the specificity of that of a class. So `:heading() {…}` would have a specificity of [0,1,0], whereas `section:heading() {…}` would have a specificity of [0,1,1].

Author: done

## Syntax

```css-nolint
:heading([ <An+B> | even | odd ]) {
  /* ... */
}
```

### Parameters

> The `:heading()` pseudo-class function takes a single argument that describes a pattern for matching heading element.

Author: done

#### Keyword values

- `odd`
  - : Represents the heading elements whose numeric position is odd: `<h1>`, `<h3>`, and `<h5>`.
- `even`
  - : Represents the heading elements whose numeric position is even: `<h2>`, `<h4>`, and `<h6>`.

#### Functional notation

- `<An+B>`
  - : Represents the heading elements whose numeric position matches the pattern `An+B`, for every positive integer or zero value of `n`, where:
    - `A` is an integer step size,
    - `B` is an integer offset,
    - `n` is all nonnegative integers, starting from 0.

    It can be read as the `An+B`-th element of a list. The `A` and `B` must both have {{cssxref("&lt;integer&gt;")}} values.

## Examples

### Keyword example

In this example the `odd` keyword is used target the `<h1>`, `<h3>`, and `<h5>` elements while the `even` keyword is used to target the `<h2>`, `<h4>`, and `<h6>` elements.

```html
<h1>Heading 1</h1>
<h2>Heading 2</h2>
<h3>Heading 3</h3>
<h4>Heading 4</h4>

> ```

Author: done

```css
:heading(odd) {
  color: tomato;
}
:heading(even) {
  color: slateblue;
}
```

{{EmbedLiveSample("Keyword_example", "", "215")}}

### Functional notation example

```html
<h1>Heading 1</h1>
<h2>Heading 2</h2>
<h3>Heading 3</h3>
<h4>Heading 4</h4>
<h5>Heading 5</h5>
<h6>Heading 6</h6>
```

```css hidden
main {
  display: flex;
  justify-content: space-around;
}
```

The `:heading(3, 4)` matches the `<h3>` and `<h4>` elements.

```css
:heading(3, 4) {
  text-decoration: tomato wavy underline;
}

> ```

Reviewer: - How about we combine all the CSS into a single code block? And the explanatory prose could be moved to after the resultant frame.
- Also, maybe we should use a dashed or solid underline..the red squiggly (wavy underline) seemed to me a spelling error at first :) and it could distract from the original intent of the example.

Reviewer: thanks a lot!
